src/plugins/index_pattern_field_editor/public/components/field_editor_flyout_content_container.tsx

```diff
@@ -0,0 +1,205 @@
+ * to retrieve/save the field in the saved object.
+ * The <FieldEditorFlyoutContent /> component is the presentational component that won't know
+ * anything about where a field comes from and where it should be persisted.
+ */
+
+export const FieldEditorFlyoutContentContainer = ({
+  field,
+  onSave,
+  onCancel,
+  docLinks,
+  indexPatternService,
+  ctx: { indexPattern, fieldTypeToProcess, search },
+  notifications,
+  fieldFormatEditors,
+  fieldFormats,
+  uiSettings,
+  usageCollection,
+}: Props) => {
+  const fieldToEdit = deserializeField(indexPattern, field);
+  const [Editor, setEditor] = useState<React.ComponentType<FieldEditorProps> | null>(null);
+  const [isSaving, setIsSaving] = useState(false);
+
+  const saveField = useCallback(
+    async (updatedField: Field) => {
+      setIsSaving(true);
+
+      const { script } = updatedField;
+
+      if (fieldTypeToProcess === 'runtime') {
+        try {
+          usageCollection.reportUiCounter(
+            pluginName,
+            usageCollection.METRIC_TYPE.COUNT,
+            'save_runtime'
+          );
+          // eslint-disable-next-line no-empty
+        } catch {}
+        // rename an existing runtime field
+        if (field?.name && field.name !== updatedField.name) {
+          indexPattern.removeRuntimeField(field.name);
+        }
+
+        indexPattern.addRuntimeField(updatedField.name, {
+          type: updatedField.type as RuntimeType,
+          script,
+        });
+      } else {
+        try {
+          usageCollection.reportUiCounter(
+            pluginName,
+            usageCollection.METRIC_TYPE.COUNT,
+            'save_concrete'
+          );
+          // eslint-disable-next-line no-empty
+        } catch {}
+      }
+
+      const editedField = indexPattern.getFieldByName(updatedField.name);
+
+      try {
+        if (!editedField) {
+          throw new Error(
+            `Unable to find field named '${updatedField.name}' on index pattern '${indexPattern.title}'`
+          );
+        }
+
+        indexPattern.setFieldCustomLabel(updatedField.name, updatedField.customLabel);
+        editedField.count = updatedField.popularity || 0;
+        if (updatedField.format) {
+          indexPattern.setFieldFormat(updatedField.name, updatedField.format);
+        } else {
+          indexPattern.deleteFieldFormat(updatedField.name);
+        }
+
+        indexPatternService.updateSavedObject(indexPattern).then(() => {
+          const message = i18n.translate('indexPatternFieldEditor.deleteField.savedHeader', {
+            defaultMessage: "Saved '{fieldName}'",
+            values: { fieldName: updatedField.name },
+          });
+          notifications.toasts.addSuccess(message);
+          setIsSaving(false);
+          onSave(editedField);
+        });
+      } catch (e) {
+        const title = i18n.translate('indexPatternFieldEditor.save.errorTitle', {
```

Comment:
there is no `await` on promise above. 
so this `catch` isn't run in case of an update error and we get stuck in "saving" state

Reply:
I believe I resolved this but double check to be sure.